distribution/tools/server-cli/src/test/java/org/elasticsearch/server/cli/ServerCliTests.java

```diff
@@ -315,6 +321,14 @@ public void testServerExitsNonZero() throws Exception {
         assertThat(exitCode, equalTo(140));
     }
 
+    public void testSecureSettingsLoaders() throws Exception {
+        this.mockSecureSettingsLoader = true;
+        Command command = newCommand();
+        command.main(new String[0], terminal, new ProcessInfo(sysprops, envVars, esHomeDir));
+        command.close();
+        assertThat(terminal.getOutput(), Matchers.containsString("Mock secure settings loader loaded"));
+    }
```

Comment:
I think there are a couple other tests we could have:
* test an empty password
* test a non existent password
* test the password is closed
* test the SecureSettings instance loaded by the loader is closed
* test a loader not support security auto config does not call auto config (or change env)

I would also avoid asserts based on output. We can maintain a handle to the loader that is created to do assertions on it. One way would be to have the mock loader instance created here locally, and the member `mockSecureSettingsLoader` is that instance, instead of a boolean.